### PR TITLE
Pass USE_SUPPLIED_CREDS to Schannel

### PR DESCRIPTION
**Pull request by nibanks**
## Description

Windows HTTP/3 client needs to be able to tell Schannel to use supplied credentials in certain scenarios.

## Testing

Added some tests to validate parameters being supported. No tests actually added to validate the feature works yet.

### CODE REVIEW

File: src/inc/msquic.h [hunk 1]
```diff
@@ -109,6 +109,7 @@ typedef enum QUIC_CREDENTIAL_FLAGS {
     QUIC_CREDENTIAL_FLAG_IGNORE_REVOCATION_OFFLINE              = 0x00001000, // Schannel only currently
     QUIC_CREDENTIAL_FLAG_SET_ALLOWED_CIPHER_SUITES              = 0x00002000,
     QUIC_CREDENTIAL_FLAG_USE_PORTABLE_CERTIFICATES              = 0x00004000, // OpenSSL only currently
```
Comment (anrossi) [inline src/inc/msquic.h:111]: Is this comment still correct? I thought @ThadHouse did work to enable portable certs on schannel

Comment (nibanks) [reply src/inc/msquic.h:111]: I think you might be right, but I don't want to contaminate this PR with fixing.

Comment (thhous-msft) [reply src/inc/msquic.h:111]: Yeah this comment is not correct anymore and could be removed.

File: src/platform/tls_schannel.c [hunk 2]
```diff
@@ -1900,6 +1900,9 @@ CxPlatTlsWriteDataToSchannel(
             ContextReq |= ASC_REQ_MUTUAL_AUTH;
         }
     }
+    if (TlsContext->SecConfig->Flags & QUIC_CREDENTIAL_FLAG_USE_SUPPLIED_CREDENTIALS) {
```
Comment (anrossi) [inline src/platform/tls_schannel.c:1903]: likewise, update the supported flags test for schannel so this flag is tested.

**Review by anrossi — DISMISSED** [review]

File: src/inc/msquic.h (hunk 1 above)
Comment (nibanks) [reply src/inc/msquic.h]: I went ahead and made the change.

**Review by thhous-msft — APPROVED** [review]

**Review by anrossi — APPROVED** [review]

**Review by thhous-msft — APPROVED** [review]

**Pull request merged**